### PR TITLE
Add testRedirectionToDifferentHostWithHeaders

### PR DESCRIPTION
**Pull request by 8W9aG**
* Tests redirection to different hosts from a
request containing headers

### CODE REVIEW

**Comment by 8W9aG** [comment]
@rastersize @dflems @jgavris

File: SPTDataLoaderTests/SPTDataLoaderServiceTest.m [hunk 1]
```diff
@@ -433,4 +433,33 @@ - (void)testConsumptionObserverTakesIntoAccountResponseHeaders
...
+
+    __block BOOL calledCompletionHandler = NO;
+    NSURL *URL = [NSURL URLWithString:@"https://newhost"];
+    NSMutableURLRequest *urlRequest = [NSMutableURLRequest requestWithURL:URL];
+    urlRequest.allHTTPHeaderFields = @{ @"Test-Header" : @"Test-Value" };
+    [self.service URLSession:self.session
+                        task:task
+  willPerformHTTPRedirection:httpResponse
+                  newRequest:urlRequest
+           completionHandler:^(NSURLRequest *newURLRequest) {
```
Comment (rastersize) [inline SPTDataLoaderTests/SPTDataLoaderServiceTest.m]: No need to wait for this block to be executed? I.e. using a test expectation

**Comment by rastersize** [comment]
If no need to wait, :+1:

File: SPTDataLoaderTests/SPTDataLoaderServiceTest.m (hunk 1 above)
Comment (jgavris) [inline SPTDataLoaderTests/SPTDataLoaderServiceTest.m]: >This method is called only for tasks in default and ephemeral sessions. Tasks in background sessions automatically follow redirects.

**Comment by jgavris** [comment]
:+1:

**Pull request merged**